zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java

```diff
@@ -77,21 +97,21 @@ public static void persist(final Path checksumPath, final ImmutableChecksumsSFV
     }
   }
 
-  /**
-   * computes a checksum for the files, in the order they're presented
-   *
-   * @return the SfvChecksum object
-   */
-  private static SfvChecksumImpl createCombinedChecksum(final Stream<Path> files) {
-    final SfvChecksumImpl checksum = new SfvChecksumImpl();
-    files.forEachOrdered(
-        path -> {
-          try {
-            checksum.updateFromFile(path);
-          } catch (final IOException e) {
-            throw new UncheckedIOException(e);
-          }
-        });
-    return checksum;
+  private static void updateChecksum(
+      final MutableChecksumsSFV checksum,
+      final Map<String, byte[]> fullFileChecksums,
+      final Path file) {
+    final String fileName = file.getFileName().toString();
+    if (fullFileChecksums.containsKey(fileName)) {
+      final Integer sstChecksum =
```

Comment:
:upside_down_face: Either use `var` or the primitive `int` here.
:wrench: I'd move the `Integer.toUnsignedLong` to this line too. Otherwise it's not the "real" `sstChecksum`.